### ISSUE TITLE
Feedback only: Infrastructure team common issues w/ etcd?

### ISSUE THREAD

**Issue by markstgodard** [cloudfoundry-incubator/etcd-release#24]
Hi @cloudfoundry-incubator/cf-infrastructure team

This is Mark from the CF Container Networking team and (as you may know) we are currently using **flannel** as our default “batteries included” overlay network.  Flannel uses **etcd** and  I’m looking at a story in our backlog about possible failure scenarios or common issues that may arise when using an `etcd` cluster:
https://www.pivotaltracker.com/story/show/121713061    

I talked to Amit on Slack and he suggested I open a story to capture this work.

Basically, I'd like to get feedback from your team about any common issues you have faced w/ etcd.
If there are issues that we have not covered in our story, then I'd like to incorporate into our testing scenarios.

cc @Amit-PivotalLabs 

Cheers

**Comment by Amit-PivotalLabs** [comment]
Hey @markstgodard 

I spoke with a couple @cloudfoundry-incubator/cf-container-networking folks in Santa Monica to better understand your use case.

1. Do not rely on data persisting in the etcd cluster.  Of course, etcd generally *shouldn't* lose data in a production configuration, but your system to be resilient to cases where it does.  We are likely to change etcd so that rolling a 1-node (e.g. in BOSH-Lite) will definitely lose data, and the failure recovery steps recommended for etcd suggest removing data.  You can see the README for more details.
1. I'd recommend not having a separate etcd cluster, it would be nice to re-use the existing CF etcd cluster.  Users don't want to have to maintain 2 or 3 separate etcd clusters, plus a consul cluster, to run CF.
1. Your stuff should work with talking to etcd over TLS.  While TLS is optional in the current OSS CF deployment of etcd, it's strongly recommended.
1. Following the first point, if you use the shared etcd cluster, keep in mind that you'll have make sure your keys don't collide with other clients of the etcd cluster.
1. Since you are not operating your own etcd cluster, you don't need to worry about complexities around orchestrating (i.e. BOSH-ifying) etcd.
1. etcd has issues under heavy write load, and even read load when storing large amounts of data.  It sounds like you're reading/writing *very* rarely (once a day), and hardly putting any data in (one "record" per Cell, maybe at most 1MB of data) so this should not be a concern.

**Comment by markstgodard** [comment]
Thanks @Amit-PivotalLabs
